examples/cookbook/recipe_numa_allocator.cpp

```diff
@@ -0,0 +1,55 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory
+//
+// Created by David Beckingsale, david@llnl.gov
+// LLNL-CODE-747640
+//
+// All rights reserved.
+//
+// This file is part of Umpire.
+//
+// For details, see https://github.com/LLNL/Umpire
+// Please also see the LICENSE file for MIT license.
+//////////////////////////////////////////////////////////////////////////////
+#include "umpire/strategy/AllocationAdvisor.hpp"
+
+#include "umpire/Allocator.hpp"
+#include "umpire/ResourceManager.hpp"
+
+#include "umpire/util/Numa.hpp"
+
+#include "umpire/util/Exception.hpp"
+
+#include <iostream>
+
+int main(int, char**) {
+  auto& rm = umpire::ResourceManager::getInstance();
+
+  /*
+   * Get the first NUMA node on the system and set this as the
+   * location for the process.
+   */
+  auto host_nodes = umpire::numa::get_host_nodes();
+  const int node = host_nodes[0];
+  umpire::numa::run_on_node(node);
```

Comment:
Do you have to set this to allocate on a specific NUMA node?

Reply:
No, I thought it would clarify the NUMA allocator, but I can switch it to get the preferred node and allocate there instead of changing the run node.